### DISCUSSION TITLE
Using the mods folder isn't working for me - what am I doing wrong?

### DISCUSSION THREAD

**Discussion by dlundahl91** [Romanitho/Winget-Install#104] [General]
Hey smartest people I know!

I cannot get the mods folder to work for my "GoToMeeting" installation and uninstallation via winget.
Here is my Intune package contents are attached to this discussion as a .zip file.

I am using this command to install:
"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -File winget-install.ps1 -AppIDs LogMeIn.GoToMeeting

and this to uninstall:
"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -File winget-install.ps1 -AppIDs LogMeIn.GoToMeeting -Uninstall

It is not doing what I have put in the mods folder though.
I have tested the scripts from the mods folder on a windows sandbox, and they did the job as needed.
Any ideas?

Thanks so much in advance!

**Comment by Romanitho** [comment]
Hello. I don't see any mistake in your conf. @KnifMelti any idea?

Is you application installed, but mods not run? or nothing works?

**Reply by dlundahl91** [reply]
Hey Romanitho! Thanks a lot for taking time out of your day to help!

The application installs perfectly fine (however not with my notification prompt - maybe not using the AppID-installed.ps1 file?). But then when trying to uninstall it just says "Uninstalling..." forever.
I could try to use the local uninstaller with a powershell script and then just terminate the processes before calling the uninstall file.

I think the problem winget experiences is that you can't uninstall the --scope machine installed GoToMeeting while the processes are running. They have to be closed first.

That is why I have the preuninstall script terminate the processes before it moves on to uninstalling.

The winget uninstall worked in my sandbox, as long as I terminated the processes first. Now I just want it to do it in our Intune environment.

**Reply by dlundahl91** [reply]
It seems like the mods aren't running: when uninstalling, it does not kill the processes.

**Comment by dlundahl91** [comment]
!!! Sorry, I think it is me who is doing something wrong!

I will test something now and get back to you.

**Reply by Romanitho** [reply]
Ok, no worries. Let us know in case someone face your case up ;)

**Reply by dlundahl91** [reply]
Hi again Romanitho!

Thanks so much still for your perseverance and assistance. You are the best. :)

I am right now stuck with this - I can uninstall it using this command:
"
# Terminate GoToMeeting processes
$GoToMeetingProcesses = Get-Process -Name "*g2m*"
Write-Host "Terminating GoToMeeting processes."
foreach($Process in $GoToMeetingProcesses){
    Write-Host "Terminating $Process.Name"
    $Process | Stop-Process
}

Start-Process -FilePath "C:\Program Files (x86)\GoToMeeting\19950\G2MUninstall.exe" -Wait -ArgumentList "/uninstall /silent"
"

But after it is uninstalled - which it does succesfully, and removes all folders from Program Files (x86) - then it is still installed according to Winget. So if I try to re-install it, it'll say it is already installed, but it is not. I then worry our winget auto-update will update the package, as it thinks it is installed. :(

Is there a way I can add something in my powershell script that removes some reg key or somewhere winget checks to see what is installed?

Thanks a ton in advance!

**Reply by dlundahl91** [reply]
Hi again @Romanitho !

Now it works!

I made a homemade uninstall script that didn't use the Winget uninstall function.
It turned out like this:
```
# Log variables
$logFolder = "C:\ProgramData\companyname"
$logName = "GoToMeeting_Uninstall_Without_Winget.log"

# Create log file and folder
# Checking if the folder already exists
If (!(Test-Path -Path $logFolder)){
    New-Item -Path $logFolder -ItemType "Directory"
}

# Start log transcript
Start-Transcript -Path "$logFolder\$logName"

# Terminate GoToMeeting processes 1st time
Write-Host "Terminating GoToMeeting processes 1st time"
$GoToMeetingProcesses = Get-Process -Name "*g2m*"
foreach($Process in $GoToMeetingProcesses){
    Sleep -Seconds 2
    Write-Host "Terminating $Process"
    $Process | Stop-Process -Force
}

# Wait 5 seconds before terminating again.
Sleep -Seconds 5

# Terminate GoToMeeting processes 2nd time
Write-Host "Terminating GoToMeeting processes 2nd time"
$GoToMeetingProcesses = Get-Process -Name "*g2m*"
foreach($Process in $GoToMeetingProcesses){
    Sleep -Seconds 2
    Write-Host "Terminating $Process"
    $Process | Stop-Process -Force
}

# Allow 10 seconds to terminate processes
Write-Host "Sleeping 10 seconds."
Sleep -Seconds 10

# Double check that they are closed
if(Get-Process -Name "*g2m*"){
    Write-Host "Not prepared for uninstallation. Error occurred."
    # Stop log transcript
    Stop-Transcript
    Exit 1
}

Write-Host "Prepared for uninstallation."

# Removing desktop shortcut
$desktopPath = [Environment]::GetFolderPath('CommonDesktopDirectory')
if(Test-Path "$desktopPath\GoTo Meeting.lnk"){
    Write-Host "Removing GoTo Meeting desktop shortcut."
    Remove-Item -Path "$desktopPath\GoTo Meeting.lnk" -Force
}

Write-Host "Starting uninstallation on path: C:\Program Files (x86)\GoToMeeting\19950\G2MUninstall.exe"

# Stop log transcript
Stop-Transcript

Start-Process -FilePath "C:\Program Files (x86)\GoToMeeting\19950\G2MUninstall.exe" -Wait -ArgumentList "/uninstall /silent"
```
The pesky processes cause the program to be hard to uninstall, and after you close them they re-appear once! So I had to terminate them twice.

Thanks so much for your time @Romanitho !!
